modules/tiles/src/tileset/tileset-3d.js

```diff
@@ -219,35 +221,113 @@ export default class Tileset3D {
     return `${tilePath}${this.queryParams}`;
   }
 
-  update(viewport) {
-    // TODO: update traverser to work with multiple viewports. Take viewport[0] for now
-    if (viewport instanceof Array) {
-      viewport = viewport[0];
+  /**
+   * Update visible tiles relying on a list of deck.gl viewports
+   * @param {WebMercatorViewport} viewports - list of deck.gl viewports
+   * @return {void}
+   */
+  update(viewports) {
+    if (this.travererseCounter > 0) {
+      return;
+    }
+    if (!(viewports instanceof Array)) {
+      viewports = [viewports];
     }
+
     this._cache.reset();
     this._frameNumber++;
-    this._frameState = getFrameState(viewport, this._frameNumber);
-    this._traverser.traverse(this.root, this._frameState, this.options);
+    this.traverseCounter = viewports.length;
+    // First loop to decrement traverseCounter
+    for (const viewport of viewports) {
+      const id = viewport.id;
+      this._initTraverser(id);
+    }
+
+    // Second loop to traverse
+    for (const viewport of viewports) {
+      const id = viewport.id;
+      if (!this.roots[id]) {
+        this.roots[id] = this._initializeTileHeaders(this.tileset, null, this.basePath);
+      }
+
+      if (!this._traversers[id]) {
+        continue; // eslint-disable-line no-continue
+      }
+      const traverser = this._traversers[id];
+      const frameState = getFrameState(viewport, this._frameNumber);
+      traverser.traverse(this.roots[id], frameState, this.options);
+    }
+  }
+
+  /**
+   * Create a traverser for a viewport if necessary
+   * @param {string} viewportId - id of a viewport
+   * @return {TilesetTraverser}
+   */
+  _initTraverser(viewportId) {
+    let traverserId = viewportId;
+    if (this.options.viewportTraversersMap) {
```

Comment:
What is the `viewportTraversersMap`? Is that a way to filter out viewports?

Maybe a cleaner API would be to just take an `ignoreViewportIds: string[]`?

Or maybe even better, simply not pass those viewports in?

Reply:
Example:
1. We have viewports `A` and `B` which have frameStates `a` and `b`;
2. Default case: tileset do traverse for `a` and tiles output on `A`, traverse for `b` is shown on `B`;
3. If we want to show traversal of `a` on both viewports, we need this mapping;
```
{
  main: 'main',
  minimap: 'main'
}
```
4. In the tile we save `viewportIds` array to check which viewports this tile should be shown on